src/reflect/scala/reflect/api/Trees.scala

```diff
@@ -2534,8 +2542,15 @@ trait Trees { self: Universe =>
     /** The underlying tree copier. */
     val treeCopy: TreeCopier = newLazyTreeCopier
 
-    /** The current owner symbol. */
-    protected[scala] var currentOwner: Symbol = rootMirror.RootClass
+    /** The current owner symbol.
+      *
+      * Access from multiple threads was reported by umad.
+      * That possibly could be solved by ensuring that every unit operates on it's own copy of the tree,
```

Comment:
Maybe a transformer captures the currentUnit, or thread, and asserts that it is local
Maybe this is a followup investigation though